### PR TITLE
feat(tasks): look up tasks by orgname

### LINKED ISSUE #11822: get tasks rest api : `org` query param is being ignored.
Whether or not you are using a valid org id in `org` query param, the get tasks request will all ways return all of the tasks.

- Create tasks for different orgs using  `influx` command:
`./influx task create 'option task = {name:"new org",every:1h} from(bucket:"b_1") |> range(start: -1h)' --org-id "0364bd08a56fa000" -t "OuNJYaSH5Vqdy3Jd8aZ7ID9C3VKynUket8EmzLR3ILNcJpvbjRZFkt1DDnFNoAs1BYEGUuG-zIPUvxYSxVidzA=="`

- Get all of the tasks using `influx` command:
`./influx task find -t "OuNJYaSH5Vqdy3Jd8aZ7ID9C3VKynUket8EmzLR3ILNcJpvbjRZFkt1DDnFNoAs1BYEGUuG-zIPUvxYSxVidzA=="`

The above command will return all of the tasks for all of the orgs:
```
ID			Name		OrganizationID		Organization	Status	Every	Cron
0364b59ea340e000	n_1		0364b4fc0d6fa000	test_1		active	1h0m0s	
0364b5e11600e000	n_1		0364b4fc0d6fa000	test_1		active	1h0m0s	
0364b5f13400e000	n_2		0364b4fc0d6fa000	test_1		active	1h0m0s	
0364b5f7ea00e000	n_3		0364b4fc0d6fa000	test_1		active	1h0m0s	
0364bd37ca40e000	new org		0364bd08a56fa000	test_2		active	1h0m0s	
0364bd40b980e000	new org 1	0364bd08a56fa000	test_2		active	1h0m0s	
0364bd4679c0e000	new org 2	0364bd08a56fa000	test_2		active	1h0m0s	
```

- Use rest api to get tasks for one of the organizations:
`curl -GET "http://localhost:9999/api/v2/tasks?org='0364bd08a56fa000'" -H "Authorization:Token OuNJYaSH5Vqdy3Jd8aZ7ID9C3VKynUket8EmzLR3ILNcJpvbjRZFkt1DDnFNoAs1BYEGUuG-zIPUvxYSxVidzA=="`

It will return tasks for all of the orgs.

### PR DESCRIPTION
**Pull request by docmerlin**
closes #11822 and #11781

This allows lookups by org name and makes the api more consistent.

### CODE REVIEW

File: http/task_service.go [hunk 1]
```diff
@@ -334,12 +334,18 @@ func decodeGetTasksRequest(ctx context.Context, r *http.Request) (*getTasksReque
 		req.filter.After = id
 	}
 
-	if orgID := qp.Get("organization"); orgID != "" {
-		id, err := platform.IDFromString(orgID)
+	if orgName := qp.Get("org"); orgName != "" {
+		o, err := orgs.FindOrganization(ctx, platform.OrganizationFilter{Name: &orgName})
```
Comment (goller) [inline http/task_service.go]: I think there needs to be some way to obfuscate the error returned here, so, it is not possible to fish for organization names.

Perhaps the API should return an empty list of tasks of the org name is unknown?

File: http/task_service.go [hunk 2]
```diff
@@ -1184,7 +1190,7 @@ func (t TaskService) FindTasks(ctx context.Context, filter platform.TaskFilter)
 		val.Add("after", filter.After.String())
 	}
 	if filter.Organization != nil {
-		val.Add("organization", filter.Organization.String())
```
Comment (goller) [inline http/task_service.go]: Perhaps add name to TaskFilter allowing one to search for tasks with name?  (could be follow-on work)

File: task_test.go [hunk 3]
```diff
@@ -64,4 +64,21 @@ from(bucket:"x")
 			t.Fail()
 		}
 	})
+	t.Run("test add new option", func(t *testing.T) {
```
Comment (goller) [inline task_test.go]: Is this test related to the new code?  Perhaps add a test using org name if it doesnt exist?

File: http/task_service.go (hunk 1 above)
Comment (mark-rushakoff) [reply http/task_service.go]: I think in other places (possibly only in some wip), I've used the phrase "unauthorized or unknown" to keep it vague.

File: task_test.go (hunk 3 above)
Comment (mark-rushakoff) [reply task_test.go]: Same question, this test doesn't seem related to this PR.

File: http/task_service.go [hunk 4]
```diff
@@ -322,7 +322,7 @@ type getTasksRequest struct {
 	filter platform.TaskFilter
 }
 
-func decodeGetTasksRequest(ctx context.Context, r *http.Request) (*getTasksRequest, error) {
+func decodeGetTasksRequest(ctx context.Context, r *http.Request, orgs platform.OrganizationService) (*getTasksRequest, error) {
```
Comment (mark-rushakoff) [inline http/task_service.go]: @lyondhill you moved some org service calls into the platform adapter in #12004. Does it make sense to move this into the platform adapter too, or should it stay here?

Comment (lyondhill) [reply http/task_service.go]: TaskService `FindTasks` currently doesn't support org name. It can be moved to the platform adapter if we wanted to update the FindTasks method in the interface. 
Out of this context The FindTasks probably needs to be updated to use `FindOptions` sometime. When that happens would be a good time to move this out of the http handler.

File: task_test.go (hunk 3 above)
Comment (docmerlin) [reply task_test.go]: Whoops, is related to another PR, the files got mixed.

File: http/task_service.go (hunk 2 above)
Comment (docmerlin) [reply http/task_service.go]: Yah, I think something like that would be better one in follow on.

File: task_test.go (hunk 3 above)
Comment (docmerlin) [reply task_test.go]: Its still a useful test, and it didn't get included in the merged pr it was supposed to, so I'll leave it in?

File: http/task_service_test.go [hunk 5]
```diff
@@ -38,6 +39,20 @@ func NewMockTaskBackend(t *testing.T) *TaskBackend {
...
 				org := &platform.Organization{}
 				if filter.Name != nil {
+					if *filter.Name == "non-existant-org" {
+						return nil, &platform.Error{
+							Err:  errors.New("org not found or unauthorized"),
+							Msg:  "org " + *filter.Name + " not found or unauthorized",
+							Code: platform.ENotFound,
+						}
+					}
+					if *filter.Name == "unauthorized" {
```
Comment (mark-rushakoff) [inline http/task_service_test.go]: It doesn't look like the `*filter.Name == "unauthorized"` path is exercised. It should be exercised or removed.

File: task_test.go [hunk 6]
```diff
@@ -64,4 +64,21 @@ from(bucket:"x")
...
+			t.Fatal(err)
+		}
+
+		op, err := options.FromScript(*tu.Flux)
+		if err != nil {
+			t.Error(err)
+		}
+		if op.Offset != 30*time.Second {
+			t.Logf("expected every to be 10s but was %s", op.Every)
+			t.Fail()
```
Comment (mark-rushakoff) [inline task_test.go]: `t.Logf; t.Fail` can be just `t.Fatalf`. And it looks like the message should read 30s.

File: http/task_service.go (hunk 2 above)
Comment (docmerlin) [reply http/task_service.go]: Here's the issue for it:
https://github.com/influxdata/influxdb/issues/12170
